tests/backends.rs

```diff
@@ -0,0 +1,82 @@
+use std::process::Command;
+
+#[test]
+fn test_backends() {
+    // Pick an arbitrary platform where linux-raw is enabled by default and
+    // ensure that the use-default crate uses it.
+    #[cfg(all(target_os = "linux", target_arch = "aarch64"))]
+    {
+        assert!(
+            !has_dependency("test-crates/use-default", &[], &["RUSTFLAGS"], "libc"),
+            "use-default depends on libc"
+        );
+        assert!(
+            has_dependency(
+                "test-crates/use-default",
+                &[],
+                &["RUSTFLAGS"],
+                "linux-raw-sys"
+            ),
+            "use-default does not depend on linux-raw-sys"
+        );
+    }
+
+    #[cfg(windows)]
+    let libc_dep = "winapi";
+    #[cfg(unix)]
+    let libc_dep = "libc";
+
+    // Test the use-libc crate, which enables the "use-libc" cargo feature.
+    assert!(
+        has_dependency("test-crates/use-libc", &[], &["RUSTFLAGS"], libc_dep),
+        "use-libc doesn't depend on {}",
+        libc_dep
+    );
+
+    // Test the use-default crate with --cfg=rustix_use_libc
+    assert!(
+        has_dependency(
+            "test-crates/use-default",
+            &[("RUSTFLAGS", "--cfg=rustix_use_libc")],
+            &[],
+            libc_dep
+        ),
+        "use-default with --cfg=rustix_use_libc does not depend on {}",
+        libc_dep
+    );
+    assert!(
+        !has_dependency(
+            "test-crates/use-default",
+            &[("RUSTFLAGS", "--cfg=rustix_use_libc")],
+            &[],
+            "linux-raw-sys"
+        ),
+        "use-default with --cfg=rustix_use_libc depends on linux-raw-sys"
+    );
+}
+
+fn has_dependency(
+    dir: &str,
+    envs: &[(&str, &str)],
+    remove_envs: &[&str],
+    dependency: &str,
+) -> bool {
+    let mut command = Command::new("cargo");
+
+    command
+        .arg("tree")
+        .arg("--quiet")
+        .arg(&format!("--invert={}", dependency))
+        .current_dir(dir);
+
+    for (key, value) in envs {
+        command.env(key, value);
+    }
+    for key in remove_envs {
+        command.env_remove(key);
+    }
+
+    let child = command.output().unwrap();
+
+    child.status.success() && !child.stdout.is_empty()
```

Comment:
My default is to propagate errors; swallowing them should IMO always deserve a comment.

In this case, any reason not to:
```
if !child.status.success() {
  panic!("Failed to invoke cargo")
}
```
at least?

Reply:
It turns out `cargo tree` fails with a non-zero status sometimes, and with just printing nothing other times. So we have to check for both. I've now added a comment about this.